roles/network_plugin/calico/tasks/install.yml

```diff
@@ -181,10 +181,31 @@
           "logSeverityScreen": "Info",
           "nodeToNodeMeshEnabled": {{ nodeToNodeMeshEnabled|default('true') }} ,
           "serviceExternalIPs": {{ _service_external_ips|default([]) }},
+          "serviceClusterIPs": [{"cidr": {{ kube_service_addresses }} }],
           "asNumber": {{ global_as_num }} }}
   changed_when: false
   when:
     - inventory_hostname == groups['kube-master'][0]
+    - not calico_no_global_as_num|default(false)
+
+- name: Calico | Set BGP Configuration without global as number defined
```

Comment:
I think maybe this task can be merged with the previous one by evaluating the condition the the spec definition:
```
"spec": {
          {% if calico_no_global_as_num|default(false) %}"asNumber": {{ global_as_num }},{% endif %}
          "serviceClusterIPs": [{"cidr": {{ kube_service_addresses }} }],
          "serviceExternalIPs": {{ _service_external_ips|default([]) }} }
```
Note that I moved the asnumber definition above to avoid having to deal with the trailing comma on the last key of the spec dict